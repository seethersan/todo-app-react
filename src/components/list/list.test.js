import React from 'react';
import { shallow, mount, render } from 'enzyme';
import List from './list.component';

window.matchMedia = window.matchMedia || function() {
    return {
        matches: false,
        addListener: function() {},
        removeListener: function() {}
    };
};

describe('expect to render List component', () => {
    var mockState= {
        tasks: [
            {
                id: 0,
                title: 'Task 1',
                isDone: true
            },
            {
                id: 1,
                title: 'Task 2',
                isDone: false
            }
        ],
        currentView: 'done'
    }

    it('expect to render List component', () => {
        expect(shallow(<List />)).toMatchSnapshot();
    });
    
    it('correctly show done tasks', () => {
        const wrapper = mount(<List {...mockState} />);
        expect(wrapper.find('Item.done').length).toEqual(1);
    });

    it('correctly show todo tasks', () => {
        mockState.currentView = 'todo';
        const wrapper = mount(<List {...mockState} />);
        expect(wrapper.find('Item.todo').length).toEqual(1);
    });
})
