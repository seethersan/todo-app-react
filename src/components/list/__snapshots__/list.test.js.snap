// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render List component expect to render List component 1`] = `ShallowWrapper {}`;
